src/lib/storage/table.hpp

```diff
@@ -134,6 +134,17 @@ class Table : private Noncopyable {
    */
   TableType get_type() const;
 
+  std::vector<std::vector<ColumnID>> get_column_ids_of_indexes() const;
+
+  template <typename Index>
+  void create_index(const std::vector<ColumnID>& column_ids) {
```

Comment:
This doesn't seem to be the job of the table.

Reply:
Since we said the table should know about its indexes, should another entity create indexes on chunks and afterwards inform the table about the indexed columns?

Reply:
I would say so. Similar to how we moved the dictionary compression to its own entity. But then, creating the index seems to be less complex, because all is hidden in the index implementations. Maybe we can leave it for now.

Reply:
Let's leave it for now and create an issue to build an index creating entity?